docs/source/format/CDataInterface/PyCapsuleInterface.rst

```diff
@@ -142,6 +154,22 @@ Arrays and record batches (contiguous tables) can implement the method
         respectively. The schema capsule should have the name ``"arrow_schema"``
         and the array capsule should have the name ``"arrow_array"``.
 
+Libraries supporting the Device interface can implement a ``__arrow_c_device_array__``
+method on those objects, which works the same as ``__arrow_c_array__`` except
+for returning a ArrowDeviceArray structure instead of a ArrowArray structure:
+
+.. py:method:: __arrow_c_device_array__(self, requested_schema: object | None = None) -> Tuple[object, object]
```

Comment:
I see we already did it above, but it's not useful to add machine-oriented type annotations to a human-readable doc. The parameter types are described explicitly below.

The HTML rendering is not terrible but it's not great either, as the signature looks crowded: https://arrow.apache.org/docs/format/CDataInterface/PyCapsuleInterface.html#arrowarray-export

Reply:
cc @wjones127

Reply:
I'm fine removing them. In general, I like having the type hints as they make signatures easy to understand from my human perspective. They can be less ambiguous than a description of a type. But given we don't have a `PyCapsule` type, I agree they don't add much value here.

Reply:
We also still have the type hints version in the https://arrow.apache.org/docs/format/CDataInterface/PyCapsuleInterface.html#protocol-typehints section a bit below

Will remove them here.